### PR TITLE
Update/tasks page

### PR DESCRIPTION
**Pull request by falklinus**
## What?
- Refactor use tasks for maintainability
- Show better count numbers in filter buttons
- Give feedback when checking in and out

## Why?
- Low effort to improve and makes it easier to work with in future

## If applicable, rooms for improvement?
- 

## Optional screenshots

https://user-images.githubusercontent.com/33982429/222396184-6fc96e87-ed0e-4334-bf19-008c61189df3.mov

## Optional checklist
- [x] Codescouted
- [ ] Unit tests written
- [x] Tested locally

### CODE REVIEW

File: src/portals/hope/features/tasks/hooks/use-tasks.ts [hunk 1]
```diff
@@ -139,37 +130,35 @@ export const useTasks = (
...
+  }, [tasksByUser])
+
+  const tabTaskCount = useMemo(() => {
+    return Object.values(TaskMode).reduce((tabTaskCount, tab) => {
+      tabTaskCount[tab] = getTasks(tab, email, tasksByUser, tasks).length
+      return tabTaskCount
+    }, {} as Record<TaskMode, number>)
+  }, [email, tasks, tasksByUser])
+
+  const currentTasks = useMemo(() => {
```
Comment (vonElfvin) [inline src/portals/hope/features/tasks/hooks/use-tasks.ts:159]: useMemo 🤝

File: src/portals/hope/features/tasks/hooks/use-tasks.ts [hunk 2]
```diff
@@ -288,40 +275,57 @@ export const useTasks = (
...
         return (
           sortDirection * stringSortByNumberOrText(a.createdAt, b.createdAt)
         )
       }
       return 0
     })
-  }, [filteredByResourceTasks, sortDirection, sortField])
+  }, [filteredByNewOrOngoingTasks, sortDirection, sortField])
+
+  const [newTasksCount, ongoingTasksCount] = filteredByResourceTasks.reduce(
```
Comment (vonElfvin) [inline src/portals/hope/features/tasks/hooks/use-tasks.ts:309]: `useMemo`?

File: src/portals/hope/pages/tasks/TasksPage.tsx [hunk 3]
```diff
@@ -759,7 +737,38 @@ const TasksPageInner: React.FC = () => {
...
+                          style={{
+                            width: '100%',
+                            padding: '4rem',
+                            display: 'grid',
+                            placeItems: 'center',
+                            gap: '0.5rem',
+                          }}
+                        >
+                          <Spinner />
+                          Moving your tasks...
```
Comment (vonElfvin) [inline src/portals/hope/pages/tasks/TasksPage.tsx:768]:
```suggestion
                          Unassigning your tasks...
```

File: src/portals/hope/pages/tasks/TasksPage.tsx [hunk 4]
```diff
@@ -759,7 +737,38 @@ const TasksPageInner: React.FC = () => {
                   </SortContainer>
                 )}
                 <ListContainer>
-                  {currentTasks
+                  {mode === TaskMode.User && checkedIn
```
Comment (vonElfvin) [inline src/portals/hope/pages/tasks/TasksPage.tsx:740]: Would this mean that you get a spinner if you don't have any tasks according to your preferences? Maybe you can test by setting the Area "Handover" or something

**Review by vonElfvin — APPROVED** [review]
Najs!

File: src/portals/hope/pages/tasks/TasksPage.tsx (hunk 4 above)
Comment (falklinus) [reply src/portals/hope/pages/tasks/TasksPage.tsx:740]: yes, i think so. I think one could improve it by having the spinner for ~15 sek, and then turn it to "No tasks matching your preferences. Your list will update when there are tasks assigned to you" or something similar

File: src/portals/hope/features/tasks/hooks/use-tasks.ts (hunk 2 above)
Comment (falklinus) [reply src/portals/hope/features/tasks/hooks/use-tasks.ts:309]: You could have it, but I don't think the computation is very costful so it wouldn't improve that much.

Comment (vonElfvin) [reply src/portals/hope/features/tasks/hooks/use-tasks.ts:309]: Ok, are the others more expensive? Or why do we mix

File: src/portals/hope/pages/tasks/TasksPage.tsx (hunk 4 above)
Comment (vonElfvin) [reply src/portals/hope/pages/tasks/TasksPage.tsx:740]: That makes sense to me

File: src/portals/hope/features/tasks/hooks/use-tasks.ts (hunk 2 above)
Comment (falklinus) [reply src/portals/hope/features/tasks/hooks/use-tasks.ts:309]: My own rule is kind of that I use it if:
- The computation is heavy
or
- It is a non-primitive that I want to pass along to something else

**Pull request merged**